### ISSUE TITLE
IFAA

### ISSUE THREAD

**Issue by quranwu** [Bioconductor/Contributions#2651]
Update the following URL to point to the GitHub repository of
the package you wish to submit to _Bioconductor_

- Repository: https://github.com/quranwu/IFAA

Confirm the following by editing each check box to '[x]'

- [x] I understand that by submitting my package to _Bioconductor_,
  the package source and all review commentary are visible to the
  general public.

- [x] I have read the _Bioconductor_ [Package Submission][2]
  instructions. My package is consistent with the _Bioconductor_
  [Package Guidelines][1].

- [x] I understand Bioconductor [Package Naming Policy][9] and acknowledge
  Bioconductor may retain use of package name.

- [x] I understand that a minimum requirement for package acceptance 
  is to pass R CMD check and R CMD BiocCheck with no ERROR or WARNINGS. 
  Passing these checks does not result in automatic acceptance. The 
  package will then undergo a formal review and recommendations for 
  acceptance regarding other Bioconductor standards will be addressed.

- [x] My package addresses statistical or bioinformatic issues related
  to the analysis and comprehension of high throughput genomic data.

- [x] I am committed to the long-term maintenance of my package. This
  includes monitoring the [support site][3] for issues that users may
  have, subscribing to the [bioc-devel][4] mailing list to stay aware
  of developments in the _Bioconductor_ community, responding promptly
  to requests for updates from the Core team in response to changes in
  _R_ or underlying software.
  
- [x] I am familiar with the [Bioconductor code of conduct][7] and 
  agree to abide by it.

I am familiar with the essential aspects of _Bioconductor_ software
management, including:

- [x] The 'devel' branch for new packages and features.
- [x] The stable 'release' branch, made available every six
      months, for bug fixes.
- [x] _Bioconductor_ version control using [Git][5]
  (optionally [via GitHub][6]).

For questions/help about the submission process, including questions about
the output of the automatic reports generated by the SPB (Single Package
Builder), please use the #package-submission channel of our Community Slack.
Follow the link on the home page of the [Bioconductor website][8] to sign up.

[1]: https://contributions.bioconductor.org/
[2]: https://bioconductor.org/developers/package-submission/
[3]: https://support.bioconductor.org
[4]: https://stat.ethz.ch/mailman/listinfo/bioc-devel
[5]: http://bioconductor.org/developers/how-to/git/
[6]: http://bioconductor.org/developers/how-to/git/sync-existing-repositories/
[7]: https://bioconductor.org/about/code-of-conduct/
[8]: https://bioconductor.org/
[9]: https://bioconductor.org/developers/package-submission/#naming

**Comment by quranwu** [comment]
Hi, For the newly submitted package, there're two things I would like to mention: 
1. The package exists on CRAN right now, but we will remove it once accepted by Bioconductor. 
2. We include set.seed() in the function for the reasons below: 
  2.1 The seed is used to reproduce the exact results under potential randomness. 
2.2 Set seed in global environment cannot be passed to the parallel computing environment (e.g., foreach function). 
2.3 The seed can be removed by set it be NULL if the user does not want to set seed.

Thanks for your time.

**Comment by quranwu** [comment]
Hi, I just modified some terms in biocViews.

**Comment by vjcitn** [comment]
Your package uses no Bioconductor infrastructure.  Please examine what is available for microbiome analysis, understand the approach to binding metadata and data together in structures like SummarizedExperiment, and let us know when you have made some decisions/revisions.  Thanks!

**Comment by vjcitn** [comment]
Your package uses no Bioconductor infrastructure.  Please examine what is available for microbiome analysis, understand the approach to binding metadata and data together in structures like SummarizedExperiment, and let us know when you have made some decisions/revisions.  Thanks!

**Comment by quranwu** [comment]
Hi, this package is intended for association identification of absolute abundance in microbiome analysis. It's similar to ANCOM or DESeq available in Bioconductor, the input is covariate of interest and absolute abundance of each taxon, and output is to find which taxon are associated with the covariates of interest. Thus, we do not need additional data transformation functions such as those in Bioconductor infrastructure. Is it required that we have to involve Bioconductor infrastructure packages in order to be accepted by Bioconductor? Thanks.

**Comment by vjcitn** [comment]
Thank you for this submission.  The involvement of Bioconductor infrastructure packages in Bioconductor components is something we generally expect from packages in the ecosystem.  A quick view of your README shows
```
results <- IFAA(MicrobData = dataM,
                CovData = dataC,
                linkIDname = "id",
                testCov = c("v1", "v2"),
                ctrlCov = c("v3"),
                nRef = 3,
                paraJobs = 2,
                fdrRate = 0.25)
Once the analysis is done, you can extract the regression coefficients along with 95% confidence intervals using this command:

results$analysisResults$sig_list_each_mean
```
It is fine to use R functions with separate arguments for assay quantifications and sample characteristics, and I'm just assuming that's what `dataM` and `dataC` are respectively.  If we look at the man page for dataC we get
```
\description{
A dataset ontains 3 covariates.
}
```
with no substantive information.  What we'd propose in Bioconductor is that the assay data be strongly bound to relevant sample metadata, using something like a SummarizedExperiment or perhaps a TreeSummarizedExperiment, and that the analysis methods operate on these.  Advantages of committing to the unified data structure strategy are that 1) it becomes harder to have mismatch of columns/rows between the separate components used in your framework, and 2) it becomes easier to do high-level filtering with the unified object designs that we have promoted for 20 years.  So I understand that you do not need these strategies to have a successful R package, but if you don't want to adopt them and have users and collaborating developers benefit from them, the package should be distributed separately from Bioconductor.

**Comment by quranwu** [comment]
Hi, 

Thanks for the suggestions. I've modified the codes and make SummarizedExperiment object as our inputs. Does this change fullfill the requirements please? Thanks.

**Comment by lshep** [comment]
@HelenaLC   ignore the ERROR that the package is on CRAN for now -- the procedure will be to review and once satisfied accept. After acceptance we will wait to add to the builders until removal from CRAN.

**Comment by quranwu** [comment]
Hi, we've update the package in both origin master and git@git.bioconductor.org:packages/IFAA to pass the test. The only error right now is for the exsistence on CRAN. Thanks.

**Comment by HelenaLC** [comment]
**Major comment**

I understand you rewrote some code for `IFAA()` and `MZILN()` to accept a `SummarizedExperiment` as input as opposed to two separate tables. However, there's little sense in doing that and then having both functions return a nested list of results and metadata. Ideally, both functions should return the same class of output as they take in; if at all possible.  

I must admit that I don't understand the exact output structure well enough, so my apologies if some of the below might not be applicable, but here are some thoughts (taking `MZILN` as an example):

- The currently returned list is difficult to handle, and blows up the screen when printed to the console. It would be much preferred to have a compact, well-structured output instead.
- Overall, to my understanding, I think you could get away with a single output object that stores all results.
- `results$analysisResults$full_results$rawCount11` is a list of matrices of identical dimensions. Could these be put inside a `SummarizedExperiment` as different assays?
- `microbName` would simply be the column names of `se` and needn't be returned. Similarly, `covriateNames` are the rownames.
- Does `results$sig_results` seems to pull out data out of the matrices above, correct? If so, significance could be saved as `TRUE/FALSE` in the `rowData` of the above SE (e.g., `rowData(se)$is_significant = TRUE/FALSE`).  
Generally, note that you are currently applying a specific filtering to determine significance, and returning redundant information, but another FDR would give different hits. So I don't see why this needs to be returned as a separate output. Alternatively, you could just remove this and let the user decide how to extract the hits they consider significant (according to their choice of an FDR threshold or other criteria).
- All "simple" elements (e.g., `seed`, `nRef`, ...) in the output list could be stuck in `se`'s `metadata` as a list.

**Less major comments**

# NEWS

- [ ] Please add a `NEWS` file to track/summarize changes from each version.

# README

- [ ] Please include Bioconductor installation instructions using `BiocManager`.

# DESCRIPTION

- [ ] Under `Description:`, "the ratio of two RA is equal TO THE ratio of their AA"; and, you say twice that "high-dimensional covariates are handled with regularization".
- [ ] Consider adding a `URL:` field pointing to the code repo, and a `BugReports:` field for reporting issues (e.g., to Github)

# data/

- [ ] In the corresponding man pages, please add details on the source information of `dataC/M` or, if it is synthetic, how the data was generated.

# unit tests

- [ ] We strongly (!) encourage unit tests; there are currently none. See [here](http://bioconductor.org/developers/how-to/unitTesting-guidelines) for guidelines.

# `BiocCheck` report

- [ ] Avoid use of `sapply`; use `vapply` instead (see `BiocCheck` report)
- [ ] Avoid `1:...`; use `seq_len/along()` instead (see `BiocCheck` report)
- [ ] Please register with the Bioc support site at https://support.bioconductor.org/ (see `BiocCheck` report)

# R/

- [ ] Consider having a look at some of coding style guidelines [here](https://bioconductor.org/developers/how-to/coding-style/). While these are *recommendations*, it's nice to be consistent (e.g., `<-` for variable assignment vs. `=` for arguments; space after `,` and around logical operators; no space around argument `=`s, etc.). The code is currently extremely compact and difficult to read.
- [ ] In addition to the above, please consider adding more linebreaks, as the code is currently very hard to follow, especially when viewed online (e.g., on GitHub). The recommended maximum number of characters per line is 80. E.g., the simplest way to make the code more readable would be i) NO vertical alignment; ii) add linebreaks when there are a lot of function arguments; and iii) define more intermediate variables vs. a `functionC(functionB(functionA(...)))`-like structure.
- [ ] Consider adding a `verbose = TRUE/FALSE` argument in `MZILN()` and `IFAA()` that specifies whether or not to print progress to the console; this is often something users would like to be in control over.

# documentation

- [ ] The `@return` section of `MZILN.R` is ill-formatted due to a typo.
- [ ] The list returned by `MZILN()` has 11 elements; however, only 3 of these are described in the function's "Value" section. Some of these 11 elements again contain another list, that might contain yet another list; none of these outputs are being described. Please be complete here. 
- [ ] Analogous to the above, `IFAA()` returns a list of length 16, but only 3 elements are described in `?IFAA`.

# vignettes/

- [ ] There should be no additional files here other than the .Rmd; please remove the vignettes.pdf/html files.
- [ ] We encourage the use of `r BiocStyle::Biocpkg("BiocStyle")` for formatting with `html_document` as rendering target; see [here](http://contributions.bioconductor.org/docs.html) for instructions.
- [ ] Eventually, please remove CRAN/GitHub-- and include `BiocManager` installation instructions.
- [ ] There are some inconsistencies in the use of back-ticks to highlight/distinguish code from normal text (e.g., in lines 59+: `SummarizedExperiment` (SE) is a class; `colData` and `countData` are slots in the SE); these would be nice to resolve.

**Comment by quranwu** [comment]
Hi, Thanks for the suggestions. For the outputs, it mainly shows the parameter estimates, standard errors, and confidence invervals for the association between taxon and covariates. Also, we expects that users would be more interested on the estimates of significant taxon, or "target taxon" in the MZILN function, so that we mannuly extract them out, as you pointed out. I agreed that our outputs may contains some outputs that's not necessary. However, we hope we could remain the "sig_results", "full_results", and "targettaxa_result_list", which we believe would be the main interest of the users. In that case, it's not possible to store them into a summarizedExperiment object, since they had different dimensions. Do you think it's acceptable if we only keep the three main results, and put others into the metadata of the summarizedExperiment object, so that the final output would be a list contains two objects, with one a list contains the three main results, and another one the summarizedExperiment object I described above? 
For the remaining minor comments, I will try to incorporate on them. 
Thanks again for your time and great suggestions.

**Comment by HelenaLC** [comment]
Hm. Why not return a long-form table instead? This would be more standard and much easier to view/filter as opposed to 5 separate matrices. E.g., something like:
```
DataFrame with 177 rows and 7 columns
            tax         cov est_save_mat p_value_save_mat  CI_low_mat  CI_up_mat     se_mat
    <character> <character>    <numeric>        <numeric>   <numeric>  <numeric>  <numeric>
1     rawCount1          v1 -0.006385916                1 -0.01836937 0.00559754 0.00611401
2     rawCount1          v2  0.002389494                1 -0.00499935 0.00977834 0.00376982
3     rawCount1          v3 -0.000347677                1 -0.00736553 0.00667018 0.00358054
4     rawCount2          v1 -0.002167259                1 -0.01560189 0.01126737 0.00685440
5     rawCount2          v2 -0.002673690                1 -0.00919896 0.00385158 0.00332922
...         ...         ...          ...              ...         ...        ...        ...
173  rawCount59          v2   0.00438131                1 -0.00252647 0.01128909 0.00352438
174  rawCount59          v3  -0.00344395                1 -0.01073366 0.00384577 0.00371924
175  rawCount60          v1   0.00640573                1 -0.00537127 0.01818273 0.00600867
176  rawCount60          v2  -0.00325590                1 -0.01022372 0.00371192 0.00355501
177  rawCount60          v3   0.00326632                1 -0.00402407 0.01055670 0.00371958
```
- In any case, it would really help if the outputs were clearly described (see my comments above regarding documentation). As there are 2 main functions, both with complex outputs, I'd say it's crucial that these are well structured and described. And that a nested list with varying elements (character, matrices, lists of lists etc.) is a bit of a nightmare from a user perspective. 
- Sure, users are interested in significant results. However, this can easily be extracted from a table such as the above using basic filtering, which you could demonstrate in the examples/vignette. There is no need to have the function do this directly and return a (duplicated) subset of the data that is tied to a specific FDR.
- I still have not wrapped my head around how the outputs are structured, but surely there's lots of duplicated information and it'd be nice to come up with a cleaner (well-structured) and compact (no redundancies) solution - open to discussion how to best do that!
- As pointed out above by Vince, I believe that Bioc's `TreeSummarizedExperiment` is meant for data like this. Perhaps it would be worth having a look at that class and see if it could accommodate the different inputs/outputs.
```
# this is far to messy...
> sapply(res[-1], class)
         covriateNames        analysisResults            sig_results targettaxa_result_list 
           "character"                 "list"                 "list"                 "list" 
               testCov             microbName                  bootB                   nRef 
           "character"            "character"              "numeric"              "integer" 
                  seed          totalTimeMins 
             "numeric"              "numeric" 
             
> sapply(res$analysisResults$full_results$rawCount11, dim)
     est_save_mat p_value_save_mat CI_low_mat CI_up_mat se_mat
[1,]            3                3          3         3      3
[2,]           59               59         59        59     59

# sig_results seems to duplicate what's in analysisResults$sig_results?   
> identical(
+   res$sig_results$rawCount11$v1,
+   res$analysisResults$sig_results$rawCount11$v1)
[1] TRUE
```

**Comment by quranwu** [comment]
Hi, we've just uploaded an update version of our package. The main updates is on outputs. Now the output is a summarizedExperiment object as suggested. The results were also wrapped up into a large matrix as suggested and stored into assay. We add a significant indicator for indicate significance. Does this output meet the requirements? We will proceed with the other comments once the output part is done. Thanks.

**Comment by HelenaLC** [comment]
Thank you for trying to address this. ...I think there has been a misunderstanding or perhaps my comments above where contradictory/confusing; apologies for that.

Similar to other statistical methods/tests, in their core, `MZLIN()` and `IFAA()` returns a table of results by (`ref_tax`), `taxon` and `cov`. It is of course fine to simply return these as such: a table (see my second comment above). My suggested output would be something like (using `MZLIN()` as an example, but the same should do for `IFAA()`):

```r
# current:
library(SummarizedExperiment)
data(dataC, dataM)

se <- SummarizedExperiment(
  assays = list(counts = dataM),
  colData = dataC)
  
res1 <- MZILN(
  experiment_dat = se,
  refTaxa = "rawCount11",
  allCov = c("v1", "v2", "v3"),
  fdrRate = 0.15)
  
# proposition:
res2 = list(
  table = assay(res1),
  metadata = metadata(res1))
  
> res2$table[1:5, 1:5]
     ref_tax     taxon cov      estimate      SE.est
1 rawCount11 rawCount1  v1 -0.0063859159 0.006114010
2 rawCount11 rawCount1  v2  0.0023894940 0.003769818
3 rawCount11 rawCount1  v3 -0.0003476765 0.003580539
4 rawCount11 rawCount2  v1 -0.0021672589 0.006854404
5 rawCount11 rawCount2  v2 -0.0026736904 0.003329221

> res2$metadata
$totalTimeMins
   elapsed 
0.05926667 

$seed
[1] 1

$fdrRate
[1] 0.15

$adjust_method
[1] "BY"
```

The advantage of this would be that the `.$table` is straightforward to understand, wrangle with, subset/filter, visualize etc. Might even be worth considering to return a `S4Vectors::DataFrame` (instead of a `data.frame`), as it has a nicer print method. Then, all of the results would nicely show in the console without blasting hundreds of lines.

*Note: Of course the above restructuring should be taken care of internally, i.e., the functions could directly return a list of 1) tabular results and 2) a list of metadata. 

Let me know if this makes sense or if you have other thoughts/suggestions/questions!

**Comment by quranwu** [comment]
Hi, thanks for the comments and suggestions. Basically, you're suggesting instead of using the SummarizedExperiment object as the output, I can simply output a list with two components, corresponding to the assay and metadata part in the current version of output, is that correct? Is the long format of the table satisfying? I will make it to S4Vectors::DataFrame as you suggested. Thanks.

**Comment by HelenaLC** [comment]
Yes, exactly, I'd argue that, in this way, all output will be 1) print nicely in the console without blowing it up; and, 2) a long-format table format is the most convenient to handle for wrangling (filtering, reshaping etc.) and visualisation.

**Comment by quranwu** [comment]
Hi, please see the updated version of our package. Sorry for taking so long since we update our algorithm to improve the speed. 
We've modified the output format as you suggested, and tried to incoperate most of your comments. However, there're still some left, and one major point is that we failed to perform the unit test. The main reason is that, as suggested by unit test, that it should start at the beginning when we build the package. Therefore, since we've finished our package, it's hard to look back and see what steps should be tested. Also, our output is qutie a complex output, and I believe unit test should be conducted directly on the final output. 
Please let me know if you have further suggestions or comments. Thanks so much for your help.

**Comment by HelenaLC** [comment]
Looking great I think! Thanks for the effort in restructuring and documenting the outputs of `IFAA` and `MZILN`. It's much clearer to me now (and hopefully future users) what's being returned, as well as the results handling (extraction, filtering etc.) being very straightforward as everything's in a `data.frame` now. Few (mostly minor) comments left:

# `BiocCheck` report

- Please try to clear as many NOTES and warnings as possible;  
the report is looking quite messy. I'd recommend checking the report  
and re-generating it as you go for full details; here some hints:
- Use \\eqn{} for on-line and \\deqn{} for displayed equations  
(analogous to Latex' single and double dollar sign)
- For single backslash commands, I believe that  
"unknown macro '\\x'" can be omitted by using '\\x{...}'
- In *metaData.R* and *runBootLassHDCI.R*, there's no need for  
`paste(0)` in `message/warning`; this is done automatically
- Avoid using `=` for assignment and use `<-` instead

# `data/` documentation

- As mentioned previously, in the corresponding man pages,  
please add details on the source information of `dataC/M` or,  
if it is synthetic, how the data was generated.  
This has still not been addressed.

# unit tests

- I don't agree that it "since [you]'ve finished our package,  
it's hard to look back and see what steps should be tested.  
Also, our output is quite a complex output, and I believe  
unit test should be conducted directly on the final output."  
Unit tests can (and should) be written for any stage in the analysis,  
including helper functions, intermediate computations etc.  
and at the very last for the final output (not vice versa).  
This is crucial in assuring that future updates (both on your end  
but also by other developers) will i) not alter results and  
ii) cause failures that might go undetected when just running  
the documentation examples in the `R CMD/BiocCheck`s.
- I highly (!) encourage reconsidering this. There are many resources online  
on how to implement unit tests. As a basic example, these can be as simple as  
checking that the output of `IFAA` is a list and the the output `data.frame`  
in `.$full_results` as the expected dimensions, column names and types etc.;  
or that some random inputs cause an error when running a certain functions.

**Comment by quranwu** [comment]
Hi,
We've submitted another version to address some notes. 
We added the unit test. 
We deleted the dataM and dataC, but generate the data itself in the examples. 
For the macro notes, it seems to be something related to loadmathjax package. The message is from IFAAfunc() and MZILN() functions, where we're using \x{} for the loadmathjax package. The mannual generated from the two functions are fine and the macros were correctly taken care of.
Thanks for all the suggestions and comments.

**Comment by HelenaLC** [comment]
- We added the unit test.  
**Thanks, greatly appreciated!** Coverage is now at >60% according to `devtools::test_coverage()` (see screenshot below). It would be nice to get this to, say, 90%+ (e.g., one of the main functions `MZILN` is currently untested).

<img width="633" alt="image" src="https://user-images.githubusercontent.com/14542264/192716505-03d38815-f275-46f4-8aeb-68e06c16e7ec.png">

* We deleted the `dataM` and `dataC`, but generate the data itself in the examples.  
**Okay, thanks!**
* For the macro notes, it seems to be something related to `loadmathjax` package. The message is from `IFAAfunc()` and `MZILN()` functions, where we're using `\x{}` for the `loadmathjax` package. The manual generated from the two functions are fine and the macros were correctly taken care of.  
**Yes, it is rendered correctly now, thanks.**

- [ ] `Suggests: BiocStyle` should be added in the `DESCRIPTION` as it's being used in the vignette.

- [ ] Especially in the vignette, try to keep to an 80 character limit  
(as the code will otherwise be difficult to read when displayed online), and sticking to a legible (and consistent!) coding style (e.g., spaces after commas, around assignment and logical operators, but not function arguments etc.)

- [ ] Regarding `mathjaxr`-related notes in the `BiocCheck` report: I found e.g. [this package](https://github.com/Bioconductor/Contributions/issues/2384) that uses `mathjaxr`, and appears to give a clean build report. I couldn't spot a relevant difference in the `DESCRIPTION` and roxygen headers compared to `IFAA`, however; perhaps @lshep can help who reviewed `rprimer`? It would be nice to resolve this as the report gets very cluttered due to this.

**Comment by quranwu** [comment]
Hi, 

We just pushed a new version in responds to the comments. 

- For the unit test, we finally improved it to 82.51% overall. Some functions, such as IFAAfunc, metaData, still have lower coverage (around 60%) due to multiple if else conditions. Those conditions are for some error or warning messages, or for some arguments, such as `verbose`, which is not tested in regular tests. 
- `Suggests: BiocStyle` is added to description. 
- We modified the vignettee to keep a consistent coding style and less than 80 character per row. 
- For mathjax, we haven't found a good way to omit the notes in biocheck.

**Comment by HelenaLC** [comment]
- Good, coverage looks much better at ≥60% for all scripts; and thanks for adding `BiocStyle` under `Suggests:`.
- Comparing the previous and current versions highlights everything as new in the vignette (see [here](https://github.com/quranwu/IFAA/commit/0949dd105eeb85cb80fda4f332a32e59cb5fab0e#diff-023d15e342a9a1e6132463d01b988fed4cf5590481fb059e6ffc05115a852856)), so I have a hard time spotting the differences. But I trust you made some.
- Hm, I'm unfortunately at a less for further suggestions regarding mathjax-related warnings, however these do not break the build and should be acceptable as the documentation is rendered correctly regardless. Perhaps something to keep in mind and hopefully resolve in the future.

**Comment by HelenaLC** [comment]
- Good, coverage looks much better at ≥60% for all scripts; and thanks for adding `BiocStyle` under `Suggests:`.
- Comparing the previous and current versions highlights everything as new in the vignette (see [here](https://github.com/quranwu/IFAA/commit/0949dd105eeb85cb80fda4f332a32e59cb5fab0e#diff-023d15e342a9a1e6132463d01b988fed4cf5590481fb059e6ffc05115a852856)), so I have a hard time spotting the differences. But I trust you made some.
- Hm, I'm unfortunately at a less for further suggestions regarding mathjax-related warnings, however these do not break the build and should be acceptable as the documentation is rendered correctly regardless. Perhaps something to keep in mind and hopefully resolve in the future. 

Please keep in mind this from the `BiocCheck`: **`ERROR: Package must be removed from CRAN.`**

**Comment by lshep** [comment]
Yes before we add to our daily builder the package must be removed from CRAN.

**Comment by lshep** [comment]
After discussion with CRAN we decided we will process the package so there is no downtime of package availability to users. Once available on Bioconductor, CRAN should be notified to archive.  I will process packages later this week or early next as we are in the middle of the Bioconductor release.

**Comment by quranwu** [comment]
Hi, 

Thanks for accepting our package. Just want to make sure that, once our package is posted on Bioconductor, should we contact CRAN to remove package from CRAN, or it will be archived automatically? Thanks. 

Quran

**Comment by lshep** [comment]
You should reach out to CRAN; it will not be done automatically

**Comment by quranwu** [comment]
Hi, 

I got a new question regarding the R version on the depends. Right now I put `R (>= 4.2.0)` as `Depends` in the description file, but after discussing with our lab, they suggest that if we could change it to `R (>= 4.0.0)` since many people are not using the latest R, and also some clusters are using old R versions. When I make this change, biocheck will give a new note saying the R version is not latest. I was just wondering if it's OK to have that note, and use `R (>= 4.0.0)` in `Depends`? Thanks!

Quran

**Comment by lshep** [comment]
You package will be released under Bioconductor 3.16 which is only compatible with 4.2.0 and not previous versions of R.  We suggest keep a different github branch with a lower version number that you could point users with older versions of R.  And your previous package versions that are released under CRAN should be compatible with the older version of R.  We would strongly urge to keep the 4.2.0 in the Bioconductor version DESCRIPTION.

**Comment by lshep** [comment]
The master branch of your GitHub repository has been added to Bioconductor's git repository.

To use the git.bioconductor.org repository, we need an 'ssh' key to associate with your github user name. If your GitHub account already has ssh public keys (https://github.com/quranwu.keys is not empty), then no further steps are required. Otherwise, do the following:

1. [Add an SSH key to your github account][]
1. [Submit your SSH key to Bioconductor][]

See further instructions at

https://bioconductor.org/developers/how-to/git/

for working with this repository. See especially

https://bioconductor.org/developers/how-to/git/new-package-workflow/
https://bioconductor.org/developers/how-to/git/sync-existing-repositories/

to keep your GitHub and Bioconductor repositories in sync.

Your package will be included in the next nigthly 'devel' build (check-out from git at about 6 pm Eastern; build completion around 2pm Eastern the next day) at

https://bioconductor.org/checkResults/

(Builds sometimes fail, so ensure that the date stamps on the main landing page are consistent with the addition of your package). Once the package builds successfully, you package will be available for download in the 'Devel' version of Bioconductor using `BiocManager::install("IFAA")`. The package 'landing page' will be created at

https://bioconductor.org/packages/IFAA

If you have any questions, please contact the bioc-devel mailing list (https://stat.ethz.ch/mailman/listinfo/bioc-devel); this issue will not be monitored further.

[Add an SSH key to your github account]: https://help.github.com/articles/connecting-to-github-with-ssh/
[Submit your SSH key to Bioconductor]: https://git.bioconductor.org/BiocCredentials/
